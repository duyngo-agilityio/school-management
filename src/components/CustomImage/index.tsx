'use client';

import Image, { ImageProps } from 'next/image';
import { memo, useState } from 'react';

// Constants
import { NO_IMAGE, NOT_FOUND_IMAGE } from '@/constants';

const CustomImage = ({ src, alt, blurDataURL, ...rest }: ImageProps) => {
  const [fallbackSrc, setFallbackSrc] = useState(false);

  const handleOnError = () => setFallbackSrc(true);
  const altImage = src !== NOT_FOUND_IMAGE && alt ? alt : NO_IMAGE;

  return (
    <Image
      src={fallbackSrc ? NOT_FOUND_IMAGE : src}
      alt={altImage}
      blurDataURL={blurDataURL}
      placeholder="blur"
      onError={handleOnError}
      {...rest}
    />
  );
};

export default memo(CustomImage);
