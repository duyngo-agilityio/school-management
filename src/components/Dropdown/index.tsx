'use client';

import { ChangeEvent, memo } from 'react';
import isEqual from 'react-fast-compare';

// Components
import { Select, SelectProps } from '@chakra-ui/react';

export interface Option {
  value: string;
  label: string;
}

interface DropdownProps extends SelectProps {
  options: Option[];
  placeholder?: string;
  width?: string;
  onChangeValue: (value: string) => void;
}

const Dropdown = ({
  options,
  placeholder = '',
  width,
  onChangeValue,
  value,
  ...rest
}: DropdownProps) => {
  const handleSelect = (e: ChangeEvent<HTMLSelectElement>) => {
    onChangeValue(e.target.value);
  };

  return (
    <Select
      w={width}
      placeholder={placeholder}
      onChange={handleSelect}
      value={value}
      {...rest}
    >
      {options.map(({ value = '', label = '' }) => (
        <option key={value} value={value}>
          {label}
        </option>
      ))}
    </Select>
  );
};

export default memo(Dropdown, isEqual);
