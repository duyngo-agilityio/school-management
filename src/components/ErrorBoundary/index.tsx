'use client';

// Libs
import { useEffect } from 'react';

// Components
import { Button, Flex, Heading } from '@chakra-ui/react';

interface ErrorProps {
  error: Error & { digest?: string };
  reset: () => void;
}

const ErrorBoundary = ({ error, reset }: ErrorProps) => {
  const handleReset = () => {
    reset();
  };

  useEffect(() => {
    // Log the error to an error reporting service
    console.error(error);
  }, [error]);

  return (
    <Flex flexDir="column" alignItems="center" gap="50px" mt={300}>
      <Heading>Something went wrong!</Heading>
      <Button onClick={handleReset}>Try again</Button>
    </Flex>
  );
};

export default ErrorBoundary;
