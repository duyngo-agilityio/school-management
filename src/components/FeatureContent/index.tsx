// Components
import { Box, Flex, Text } from '@chakra-ui/react';

// Mocks
import { CONTENT_DASHBOARD } from '@/mocks';

const FeatureContent = () =>
  CONTENT_DASHBOARD.map(({ id, icon, title, description }) => (
    <Flex key={id} mb="50px">
      <Box w="36px" mr="20px">
        <Flex
          alignItems="center"
          justifyContent="center"
          w="36px"
          h="36px"
          bg="backgroundLinkWater"
          borderRadius="lg"
        >
          {icon}
        </Flex>
      </Box>
      <Flex flexDirection="column">
        <Text variant="primary" fontWeight="medium" size="lg" pb="16px">
          {title}
        </Text>
        <Text variant="primary" size="sm">
          {description}
        </Text>
      </Flex>
    </Flex>
  ));

export default FeatureContent;
