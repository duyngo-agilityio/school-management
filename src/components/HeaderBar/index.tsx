'use client';

import { ReactNode } from 'react';
import { useRouter } from 'next/navigation';

// Icons
import { BellNotificationIcon } from '@/icons';

// Components
import { Button, Flex, useDisclosure } from '@chakra-ui/react';
import ConfirmModal from '../Modal/ConfirmModal';

// Services
import { logout } from '@/actions';

// Constants
import { ROUTES } from '@/constants';

interface HeaderBarProps {
  children: ReactNode;
  width?: string;
}

const HeaderBar = ({ children, width }: HeaderBarProps) => {
  const router = useRouter();
  const {
    isOpen: isOpenConfirm,
    onOpen: onOpenConfirm,
    onClose: onCloseConfirm,
  } = useDisclosure();

  const handleLogout = async () => {
    await logout();
    router.replace(ROUTES.SIGN_IN);
  };

  return (
    <>
      {isOpenConfirm && (
        <ConfirmModal
          onClose={onCloseConfirm}
          onSubmit={handleLogout}
          title="Logging Out"
          subTitle="Are you sure you want to log out?"
          labelButton="Logout"
        />
      )}
      <Flex justifyContent="space-between" w={width}>
        {children}
        <Flex alignItems="center">
          <BellNotificationIcon />
          <Button ml="20px" onClick={onOpenConfirm}>
            Log Out
          </Button>
        </Flex>
      </Flex>
    </>
  );
};

export default HeaderBar;
