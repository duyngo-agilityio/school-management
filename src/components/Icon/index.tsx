import { Flex } from '@chakra-ui/react';

interface IconProps {
  icon: JSX.Element;
  width?: string;
  height?: string;
  bg?: string;
}

const Icon = ({
  icon,
  width = '36px',
  height = '36px',
  ...rest
}: IconProps) => (
  <Flex
    alignItems="center"
    justifyContent="center"
    w={width}
    h={height}
    {...rest}
  >
    {icon}
  </Flex>
);

export default Icon;
