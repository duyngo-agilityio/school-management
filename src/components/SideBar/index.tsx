'use client';

// Libs
import React from 'react';

// Components
import { Box } from '@chakra-ui/react';
import Profile from '../Profile';

const Sidebar = () => {
  return (
    <Box
      bg="backgroundPrimary"
      borderRight="1px"
      w="241px"
      pos="fixed"
      h="full"
    >
      <Box borderBottom="1px" borderBottomColor="borderSecondary">
        <Profile src="/logo.png" title="Udemy Inter. school" variant="md" />
      </Box>
      {/* TODO: Implement Nav component */}
    </Box>
  );
};

export default Sidebar;
