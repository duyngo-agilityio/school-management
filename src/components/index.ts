export { default as Profile } from './Profile';
export { default as SideBar } from './SideBar';
export { default as SearchInput } from './SearchInput';
export { default as FeatureContent } from './FeatureContent';
export { default as TableStudent } from './TableStudent';
export { default as TableTeacher } from './TableTeacher';
export { default as Dropdown } from './Dropdown';
export { default as PasswordInput } from './PasswordInput';
export { default as NotFound } from './NotFound';
export { default as HeaderBar } from './HeaderBar';
export { default as Icon } from './Icon';
export { default as StudentContainer } from './StudentContainer';
export { default as UploadImage } from './UploadImage';
