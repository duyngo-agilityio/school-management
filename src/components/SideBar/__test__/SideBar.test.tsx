import { render } from '@testing-library/react';

// Components
import { SideBar } from '@/components';

describe('SideBar Component', () => {
  it('should match snapshot for SideBar', () => {
    const { container } = render(<SideBar />);

    expect(container).toMatchSnapshot();
  });
});
