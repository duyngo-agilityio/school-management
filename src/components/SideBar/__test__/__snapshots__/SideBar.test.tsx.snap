// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component should match snapshot for SideBar 1`] = `
<div>
  <div
    class="css-6ozlle"
  >
    <div
      class="css-1xlms3t"
    >
      <a
        href="/"
      >
        <div
          class="css-0"
        >
          <div
            class="css-79elbk"
            style="width: 65px; height: 65px;"
          >
            <img
              alt="Udemy Inter. school"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Flogo.png&w=3840&q=75"
              srcset="/_next/image?url=%2Flogo.png&w=640&q=75 640w, /_next/image?url=%2Flogo.png&w=750&q=75 750w, /_next/image?url=%2Flogo.png&w=828&q=75 828w, /_next/image?url=%2Flogo.png&w=1080&q=75 1080w, /_next/image?url=%2Flogo.png&w=1200&q=75 1200w, /_next/image?url=%2Flogo.png&w=1920&q=75 1920w, /_next/image?url=%2Flogo.png&w=2048&q=75 2048w, /_next/image?url=%2Flogo.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            />
          </div>
          <p
            class="chakra-text css-10onoed"
          >
            Udemy Inter. school
          </p>
        </div>
      </a>
    </div>
    <a
      href="/"
      style="text-decoration: none;"
    >
      <div
        class="css-up6y35"
        role="group"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="css-iikgio"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 16 17"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.6666 7.49978C14.6666 6.69312 14.1266 5.65978 13.4666 5.19978L9.34659 2.31312C8.41325 1.65978 6.91325 1.69312 6.01325 2.39312L2.41992 5.19312C1.81992 5.65978 1.33325 6.65312 1.33325 7.40645V12.3464C1.33325 13.8931 2.59325 15.1598 4.13992 15.1598H11.8599C13.4066 15.1598 14.6666 13.8931 14.6666 12.3531V10.2864"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M8 12.4932V10.4932"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
          Dashboard
        </div>
      </div>
    </a>
    <a
      href="/teachers"
      style="text-decoration: none;"
    >
      <div
        class="css-up6y35"
        role="group"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="css-iikgio"
          >
            <svg
              fill="white"
              height="16"
              id="Layer_1"
              version="1.1"
              viewBox="0 0 299.97 299.97"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <g>
                  <g>
                    <path
                      d="M149.985,126.898c34.986,0,63.449-28.463,63.449-63.449C213.435,28.463,184.971,0,149.985,0S86.536,28.463,86.536,63.449     C86.536,98.436,114.999,126.898,149.985,126.898z M149.985,15.15c26.633,0,48.299,21.667,48.299,48.299     s-21.667,48.299-48.299,48.299s-48.299-21.667-48.299-48.299S123.353,15.15,149.985,15.15z"
                    />
                    <path
                      d="M255.957,271.919l-20.807-86.313c-2.469-10.244-11.553-17.399-22.093-17.399c-13.216,0-114.332,0-126.145,0     c-10.538,0-19.623,7.155-22.093,17.399l-20.807,86.313c-3.444,14.289,7.377,28.051,22.093,28.051h167.76     C248.563,299.97,259.407,286.229,255.957,271.919z M66.105,284.82c-4.898,0-8.513-4.581-7.364-9.35l20.807-86.314     c0.823-3.415,3.851-5.799,7.365-5.799H121.4l-9.553,67.577c-0.283,2,0.244,4.029,1.464,5.637l21.422,28.249H66.105z      M127.291,249.932l9.411-66.574h26.567l9.411,66.574l-22.695,29.927L127.291,249.932z M233.865,284.82h-68.628l21.421-28.248     c1.22-1.609,1.747-3.638,1.464-5.637l-9.553-67.577h34.487c3.513,0,6.542,2.385,7.365,5.8l20.807,86.313     C242.377,280.235,238.769,284.82,233.865,284.82z"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
          Teachers
        </div>
      </div>
    </a>
    <a
      href="/students"
      style="text-decoration: none;"
    >
      <div
        class="css-up6y35"
        role="group"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="css-iikgio"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.05 2.52997L4.03002 6.45997C2.10002 7.71997 2.10002 10.54 4.03002 11.8L10.05 15.73C11.13 16.44 12.91 16.44 13.99 15.73L19.98 11.8C21.9 10.54 21.9 7.72997 19.98 6.46997L13.99 2.53997C12.91 1.81997 11.13 1.81997 10.05 2.52997Z"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M5.63012 13.08L5.62012 17.77C5.62012 19.04 6.60012 20.4 7.80012 20.8L10.9901 21.86C11.5401 22.04 12.4501 22.04 13.0101 21.86L16.2001 20.8C17.4001 20.4 18.3801 19.04 18.3801 17.77V13.13"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M21.3999 15V9"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
          Students/ classes
        </div>
      </div>
    </a>
  </div>
</div>
`;
